Cache the WasmModule in a static field to avoid parsing it multiple times

chirino opened the pull request:
Follow up PR to https://github.com/dylibso/chicory/pull/1067

We use the Initialization-on-Demand Holder idiom so that we get lazy, thread-safe initialization of static value.

andreaTP commented:
@chirino This PR shouldn’t have been merged because:

- There are unresolved comments: https://github.com/dylibso/chicory/pull/1072#issuecomment-3401009315
- It wasn’t approved yet, and our contribution guidelines require at least one positive review: https://github.com/dylibso/chicory/blob/main/CONTRIBUTING.md#code-reviews

I’ve applied the suggested changes myself and opened #1076 to fix this on top.

chirino commented:
My bad, I thought https://github.com/dylibso/chicory/compare/a74c55c2eab1559e07c1ef3d614e50025f0224ce..02a28e557a74f73bf3b44e8175bb7172c0525362 was addressing the remaining issues.
